config/zephyr/Kconfig

```diff
@@ -281,6 +281,12 @@ config CHIP_PERSISTENT_SUBSCRIPTIONS
 	  subscriptions instead of waiting for the subscriber node to realize that
 	  the publisher is alive again.
 
+config CHIP_STATISTICS
```

Comment:
Shouldn't we add depends on CHIP_LIB_SHELL here? Generally I think we should add information that this is shell module, as based on Kconfig name and description it seems to be a generic module, while it cannot be used outside of the shell, right?

Reply:
In fact, it can. A user might want to enable statistics and use a 3rd-party channel to retrieve them.

Reply:
I mean that current implementation doesn't expose API to get statistics information from some 3rd party module, as the implementation is invoked from command handlers.

Reply:
So this Kconfig switch only enables macros defined in `SystemStats.h` to make sure that `System::Stats::GetResourcesInUse` and `System::Stats::GetHighWatermarks` return meaningful data. Shell is one of the consumers of these values, but one enable `CONFIG_CHIP_STATISTICS` **and** not enable shell at all, and e.g. add a custom BLE service that would expose numbers from the mentioned functions. That's why I think this configuration does not depend on the shell. Does that make sense?

Reply:
Ah I see! I missed the fact that gn flag is mapped on the `CHIP_SYSTEM_CONFIG_PROVIDE_STATISTICS` and I was looking for `CHIP_STATISTICS` uses. Yes, now it makes sense.